exoscale/api/compute.py

```diff
@@ -0,0 +1,1714 @@
+    def update(self, reverse_dns=None):
+        """
+        Update the Elastic IP properties.
+
+        Parameters:
+            reverse_dns (str): Elastic IP reverse DNS record
+
+        Returns
+            None
+        """
+
+        if reverse_dns != None:
+            try:
+                if reverse_dns == "":
+                    self.compute.cs.deleteReverseDnsFromPublicIpAddress(id=self.id)
+                else:
+                    self.compute.cs.updateReverseDnsForPublicIpAddress(
+                        id=self.id, domainname=reverse_dns
+                    )
+            except CloudStackApiException as e:
+                raise APIException(e.error["errortext"])
+
+    def attach_instances(self, instances):
+        """
+        Attach the Elastic IP to instances.
+
+        Parameters:
+            instances ([Instance]): list of instances to attach the Elastic IP to
+
+        Returns:
+            None
+        """
+
+        for instance in instances:
+            instance.attach_elastic_ip(elastic_ip=self)
+
+    def detach_instances(self, instances):
+        """
+        Detach the Elastic IP from instances it is attached to.
+
+        Parameters:
+            instances ([Instance]): list of instances to detach the Elastic IP from
+
+        Returns:
+            None
+        """
+
+        for instance in instances:
+            instance.detach_elastic_ip(elastic_ip=self)
+
+
+@attr.s
+class Instance(Resource):
+    """
+    A Compute instance.
+
+    Attributes:
+        id (str): the instance unique identifier
+        name (str): the instance hostname/display name
+        zone (Zone): the zone in which the instance is located
+        type (InstanceType): the instance type
+        template (InstanceTemplate): the instance template
+        ipv4_address (str): the instance public network interface IP address
+        ipv6_address (str): the instance public network interface IPv6 address,
+            or None if IPv6 is not enabled
+        ssh_key (SSHKey): the SSH key installed in the instance user account,
+        or None if no SSH key specified during instance creation
+    """
+
+    compute = attr.ib(repr=False)
+    res = attr.ib(repr=False)
+    id = attr.ib()
+    name = attr.ib()
+    zone = attr.ib(repr=False)
+    type = attr.ib(repr=False)
+    template = attr.ib(repr=False)
+    ipv4_address = attr.ib(repr=False)
+    ipv6_address = attr.ib(default=None, repr=False)
+    ssh_key = attr.ib(default=None, repr=False)
+
+    @classmethod
+    def from_cs(cls, compute, res):
+        return Instance(
+            compute,
+            res,
+            id=res["id"],
+            name=res["displayname"],
+            zone=compute.get_zones(id=res["zoneid"])[0],
+            type=compute.get_instance_types(id=res["serviceofferingid"])[0],
+            template=compute.get_instance_templates(id=res["templateid"])[0],
+            ipv4_address=list(i for i in res["nic"] if i["isdefault"])[0].get(
```

Comment:
Nice, thanks!

Reply:
I'm commenting only because you seem to agree with the suggested changes but they haven't been applied.

I also happen to have a soft spot for 

`next(i for i in res["nic"] if i["isdefault"])`
over
`(i for i in res["nic"] if i["isdefault"])[0]`

When trying to only get the first element of a list/generator comprehension

Reply:
Thank you for the heads up about `next()`, didn't know about it and indeed it feels better 😊 Changed in 381339e (along with the skipped suggestion from @brutasse). Too bad it doesn't work on actual lists, only iterators...